lib/main.dart

```diff
@@ -1,43 +1,56 @@
 import 'package:clnapp/api/api.dart';
 import 'package:clnapp/api/client_provider.dart';
 import 'package:clnapp/api/cln/cln_client.dart';
+import 'package:clnapp/constants/user_setting.dart';
+import 'package:clnapp/helper/settings/get_settings.dart';
 import 'package:clnapp/utils/app_provider.dart';
 import 'package:clnapp/views/app_view.dart';
 import 'package:clnapp/views/home/home_view.dart';
+import 'package:clnapp/views/setting/setting_view.dart';
 import 'package:flutter/material.dart';
 import 'package:trash_themes/themes.dart';
-import 'dart:io' show Platform;
 
 Future<void> main() async {
   var provider = await AppProvider().init();
 
-  /// TODO just for now we insert by hand GRPC information, but in the future
-  /// we need to have a Setting UI!
-  var certificateDir = Platform.environment['CLN_CERT_PATH'];
-  if (certificateDir == null) {
-    throw Exception("Please export the CLN_CERT_PATH for your system");
+  Setting setting = Setting();
+
+  await getSettingsInfo().then((value) => {
+        setting = value,
+      });
+
+  if (setting.path != "No path found") {
+    provider.registerLazyDependence<AppApi>(() {
+      if (setting.connectionType == clients[0]) {
```

Comment:
with better setting storage, you can save this if else and use the API that make the conf for you.

An example, you store in the Preference a key value JSON like

```
"clientSetting": {
     "mode": "grpc", 
    "path": "blabla"
}
```

 you look inside the ApiProvider API you have `getClient(
      {required ClientMode mode, Map<String, dynamic> opts = const {}})`

If you encode the setting in a generic way like

```
class SettingClient {
    final String modeStr;
    final HashMap opts;
}
```

You save a lot of code because when you load the setting you decode the json in  the SettingClient class and you can do

```
var setting = //load setting
var client ApiProvicer.getClient(setting.getClientModeEnum(), setting.opts);
```

Where `getClientModeEnum` is just a map
```
 Map<String, ClientMode> = {
    "grpc": ClientMode.grpc,
/// other
}
```